### ISSUE TITLE
Unable to spin up cluster with > 3 nodes.

### ISSUE THREAD

**Issue by darkryder** [Seagate/cortx-motr#893]
I am following [this guide](https://github.com/Seagate/cortx-motr/wiki/Build-Motr-from-Source-in-a-Cluster) to spin up a motr cluster. I am able to spin up a 3 node cluster, even with multiple disks and different fault tolerance strategies. 

However, as soon as I attempt to introduce more nodes to the setup, I am unable to bootstrap the cluster. I believe motr is correctly setup on those nodes since I can spin up a 3 node cluster using these newly added nodes. I've tried the following configs:

```
n_VMs   n_disksPerVM   data_units   parity_units   spare_units   status
3       1              1            2              0             works
3       1              2            1              0             works
3       2              1            2              0             works
3       4              1            2              0             works
4       2              1            2              0             fails
4       2              2            1              0             fails    
4       1              2            2              0             fails
6       1              1            2              0             fails
6       1              1            2              3             fails
6       1              4            2              0             fails
8       >= 2           8            2              0             fails (attempting to replicate https://github.com/Seagate/cortx-motr/discussions/285)
```

It fails with this message: `motr-mkfs[140435]: Motr panic: (({ unsigned __nr = (zones_nr); unsigned i; typeof(0ULL) __accum = (0ULL); for (i = 0; i < __nr; ++i) { __accum = __accum + zone_percent[i]; } __accum; }) == 100) at m0_be_allocator_create() be/alloc.c:837`

Motr git version: `bcb31b1a4c`
hare git version: `f5e0dca794a`

It fails regardless of whether I set the `allowed_failures` variable or not. I'm attaching an [example CDF](https://github.com/Seagate/cortx-motr/files/6889348/CDF-2-1.txt) for 4VMs * 3 disks each with 3-way replication. (filename changed since github doesn't allow .yaml file uploads).

Is anyone else facing the same issue?

**Comment by johnbent** [comment]
Hello @darkryder .  Thanks for opening this issue.  Are you trying to grow an existing motr cluster?  That is not currently supported so I wouldn't expect that to succeed.  If you are merely trying to create a new larger cluster using the pieces of the old one, then I would expect that to work but maybe you need to clean up or something first?  Plus @andriytk  to confirm that growing an existing cluster is not currently supported.

**Comment by andriytk** [comment]
This looks like a regression.  @darkryder could you please try `sage-base-2.0` tag of Motr+Hare? This is known to work on more than 3 nodes. It will help to figure out whether this is a regression or some testing issue. Thanks!

**Comment by darkryder** [comment]
@johnbent I am currently trying to setup a fresh cluster w/ size > 3 nodes. I am passing `--mkfs` to `hctl bootstrap` which seems to clear the data drives at the very least. I can't see any public flag to clear any other config/map state.

@andriytk Thanks for the tag! It will indeed help me figure out if there's some problem with how I'm setting up the cluster. I'll check it today.

**Comment by darkryder** [comment]
@andriytk I have been able to successfully spin up a 6 node cluster w/ 4+2 erasure coding using `sage-base-2.0`.

**Comment by andriytk** [comment]
So this is definitely a regression. Thanks for reporting!

**Comment by madhavemuri** [comment]
We are looking into the root cause of this issue.
https://jts.seagate.com/browse/EOS-16272?focusedCommentId=2154981
@yatin-mahajan  @suvratjoshi : ^

**Comment by andriytk** [comment]
Another ticket was created about this - https://jts.seagate.com/browse/EOS-23180 and started being looking at, on priority.

**Comment by madhavemuri** [comment]
@andriytk @darkryder : 
 N=1 and  K=2  works fine for dix pool for 4-nodes setup, as earlier it was N=1 and k=0 by default.

As with recent changes  K for DIX pool is taken as `number of nodes - 1` by default,  K is taken as 3 which caused mkfs issue.
I think we need to add support/fix the issue for K>2 for DIX pool.

**Comment by andriytk** [comment]
Thanks @madhavemuri for the update.

@darkryder as a workaround, you can specify the DIX pool redundancy parameters explicitly in the CDF, under the `pools:` section like this:

```
pools:
  - name: my-dix-pool
    type: dix
    data_units: 1
    parity_units: 2
    spare_units: 0
```

**Comment by darkryder** [comment]
Am I correct in assuming that a dix pool will store index-es [link](https://github.com/Seagate/cortx-motr/blob/main/motr/client.h#L365), while an sns pool will store [Objects](https://github.com/Seagate/cortx-motr/blob/main/motr/client.h#L365)?

I did want to store objects, which is why I set type to sns.

**Comment by andriytk** [comment]
> dix pool will store index-es, while an sns pool will store Objects?

@darkryder that's correct. Before, the DIX pool was always 1+0. But with the recent changes it depends on the number of configured nodes and becomes 1+(n - 1), so for 4 nodes it automatically is configured to 1+3. And there is some bug such configurations of DIX pool on 4+ nodes. We are working on this.

Keep tuned, and many thanks for trying it!

**Comment by madhavemuri** [comment]
@darkryder @andriytk : Fix for this issue is landed to main `1cb064df97d1a313eb577d942358564a2ba47c60` 
Configure DIX pool in a following way,
  - name: my-dix-pool
    type: dix
    data_units: 1   # N data units
    parity_units: 2  # K  parity units
    spare_units: 0   # S  spare units
